src/osyris/spatial/spatial.py

```diff
@@ -0,0 +1,43 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Osyris contributors (https://github.com/nvaytet/osyris)
+
+import numpy as np
+
+
+def extract_sphere(dataset, radius, origin):
+    """
+    Extract a spherical subdomain around an origin point.
+    """
+    subdomain = dataset.__class__(nout=dataset.meta["nout"], path=dataset.meta["path"])
+    subdomain.meta = dataset.meta
+    subdomain._parent = dataset
+
+    for name, group in dataset.items():
+        pos = group.get("position", group.parent["amr"]["position"])
```

Comment:
This might be ok for now, or we may consider making the search for the `position` safer.
Maybe Datagroups could have a `.kind` member, just like the loaders, that would tell us what kind of group it is (`amr`, `part` or `sink`). And based on the group kind we could safely load the amr position.

If someone makes a group that has no position, and tries to make a subdomain, it could wrongly take the positions from the amr group.

Reply:
I'm not sure how a `.kind` member solves this issue. How do we assign this member when a user creates a new group?

Reply:
Well I think the point is that we don't assign it, so that it doesn't accidentally go and find the amr position when it shouldn't? If you create a new group that has nothing to so with amr, and then make a subdomain, you will try to get the position from amr, with a different length....